controllers/controlplaneupgrade_controller.go

```diff
@@ -363,3 +367,52 @@ func getCapiMachine(ctx context.Context, client client.Client, nodeUpgrade *anyw
 	}
 	return machine, nil
 }
+
+func createKubeVipCMIfNotExist(ctx context.Context, client client.Client, cpUpgrade *anywherev1.ControlPlaneUpgrade) error {
+	kubeVipCM := &corev1.ConfigMap{}
+	if err := client.Get(ctx, GetNamespacedNameType(constants.KubeVipConfigMapName, constants.EksaSystemNamespace), kubeVipCM); err != nil {
+		if apierrors.IsNotFound(err) {
+			kubeVipCM, err = kubeVipConfigMap(cpUpgrade)
+			if err != nil {
+				return err
+			}
+			if err := client.Create(ctx, kubeVipCM); err != nil {
+				return fmt.Errorf("failed to create %s config map: %v", constants.KubeVipConfigMapName, err)
+			}
+		} else {
+			return fmt.Errorf("getting %s configmap: %v", constants.KubeVipConfigMapName, err)
+		}
+	}
+	return nil
+}
+
+func kubeVipConfigMap(cpUpgrade *anywherev1.ControlPlaneUpgrade) (*corev1.ConfigMap, error) {
+	kcpSpec, err := decodeAndUnmarshalKcpSpecData(cpUpgrade.Spec.ControlPlaneSpecData)
+	if err != nil {
+		return nil, err
+	}
+	var kubeVipConfig string
+	for _, file := range kcpSpec.KubeadmConfigSpec.Files {
+		if file.Path == kubeVipStaticPodPath {
+			kubeVipConfig = file.Content
+			break
+		}
+	}
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.KubeVipConfigMapName,
+			Namespace: constants.EksaSystemNamespace,
+			OwnerReferences: []metav1.OwnerReference{{
```

Comment:
This is good but this doesn't ensure it gets cleaned up right away right? Meaning, if a new CPUpgrade needs to get created with the same name due to some subsequent upgrade, we will never get the new CM for this if it never gets deleted?

Reply:
That's a good point. I will clean it up once all the nodes are upgraded for a given cpu. This way we ensure we always clean it after a successful upgrade.

Reply:
I think when you have `BlockOwnerDeletion` set on the child object, it blocks the parent object deletion. @rahulbabu95 you can test it out just to be sure